investments/views.py

```diff
@@ -1,3 +1,27 @@
-from django.shortcuts import render
+from django.views import View
+from django.http import JsonResponse
 
-# Create your views here.
+from .models import Investment, Grade, Image
+
+class InvestmentListView(View):
+    def get(self, request):
+        try:
+            if not Investment.objects.all():
+               return JsonResponse({'MESSAGE' : 'INVESTMENTS_DOES_NOT_EXISTS'}, status = 401)
+
+            investments = Investment.objects.all()
+
+            return JsonResponse({"investments" : [
+                {
+                    "id"               : investment.id,
+                    "name"             : investment.name,
+                    "return_rate"      : investment.return_rate,
+                    "duration"         : investment.duration,
+                    "target_amount"    : investment.target_amount,
+                    "grade"            : investment.grade.name,
+                    "image"            : investment.image_set.all().first().url,
+                    "recrutement_rate" : int(investment.current_amount/investment.target_amount*100)
+                } for investment in investments]}, status = 200)
```

Comment:
investment 모델 객체 N개에 역참조하고 있는 image에 대하여 + 1개의 쿼리가 새로 발생할 것 같습니다!! ORM 최적화를 통해서 문제를 해결해보는 것은 어떨까요?

```suggestion
            investments = Investment.objects.prefetch_related("image_set").all()

            return JsonResponse({"investments" : [
                {
                    "id"               : investment.id,
                    "name"             : investment.name,
                    "return_rate"      : investment.return_rate,
                    "duration"         : investment.duration,
                    "target_amount"    : investment.target_amount,
                    "grade"            : investment.grade.name,
                    "image"            : investment.image_set.all().first().url,
                    "recrutement_rate" : int(investment.current_amount/investment.target_amount*100)
                } for investment in investments]}, status = 200)
```

Reply:
확인 후 조언 주신대로 변경 하였습니다!!! 감사합니다.